pkg/resources/kafka/configmap.go

```diff
@@ -70,25 +70,26 @@ func (r *Reconciler) getConfigProperties(bConfig *v1beta1.BrokerConfig, id int32
 		log.Error(err, "setting zookeeper.connect parameter in broker configuration resulted an error")
 	}
 
-	// Add Cruise Control SSL configuration
+	// Add Cruise Control Metrics Reporter SSL configuration
 	if util.IsSSLEnabledForInternalCommunication(r.KafkaCluster.Spec.ListenersConfig.InternalListeners) {
 		if !r.KafkaCluster.Spec.IsClientSSLSecretPresent() {
 			log.Error(errors.New("cruise control metrics reporter needs ssl but client certificate hasn't specified"), "")
 		}
-		if err := config.Set("cruise.control.metrics.reporter.security.protocol", "SSL"); err != nil {
-			log.Error(err, "setting cruise.control.metrics.reporter.security.protocol in broker configuration resulted an error")
+		keyStoreLoc := clientKeystorePath + "/" + v1alpha1.TLSJKSKeyStore
+		trustStoreLoc := clientKeystorePath + "/" + v1alpha1.TLSJKSTrustStore
+
+		sslConfig := map[string]string{
+			"security.protocol":       "SSL",
+			"ssl.truststore.location": trustStoreLoc,
+			"ssl.keystore.location":   keyStoreLoc,
+			"ssl.keystore.password":   clientPass,
+			"ssl.truststore.password": clientPass,
 		}
-		if err := config.Set("cruise.control.metrics.reporter.ssl.truststore.location", clientKeystorePath+"/"+v1alpha1.TLSJKSTrustStore); err != nil {
-			log.Error(err, "setting cruise.control.metrics.reporter.ssl.truststore.location in broker configuration resulted an error")
-		}
-		if err := config.Set("cruise.control.metrics.reporter.ssl.truststore.password", clientPass); err != nil {
-			log.Error(err, "setting cruise.control.metrics.reporter.ssl.truststore.password parameter in broker configuration resulted an error")
-		}
-		if err := config.Set("cruise.control.metrics.reporter.ssl.keystore.location", clientKeystorePath+"/"+v1alpha1.TLSJKSKeyStore); err != nil {
-			log.Error(err, "setting cruise.control.metrics.reporter.ssl.keystore.location parameter in broker configuration resulted an error")
-		}
-		if err := config.Set("cruise.control.metrics.reporter.ssl.keystore.password", clientPass); err != nil {
-			log.Error(err, "setting cruise.control.metrics.reporter.ssl.keystore.password parameter in broker configuration resulted an error")
+
+		for k, v := range sslConfig {
+			if err := config.Set(fmt.Sprintf("cruise.control.metrics.reporter.%s", k), v); err != nil {
+				log.Error(err, fmt.Sprintf("setting cruise.control.metrics.reporter.%s parameter in broker configuration resulted an error", k))
+			}
```

Comment:
**Question**: This block looks similar to the one in `pkg/resources/cruisecontrol/configmap.go`, could we possibly extract this to a function to have a single place of definition for the logic or does it not make sense in this case?

Also `generateListenerSSLConfig()` looks somewhat of a generic variation of this, could that be used in some form or shape?

Reply:
This is not completely similar here we talking about cruise.control.metrics.reporter.%s. The sslConfig map variable can be used multiple times or with a variable in the  config.Set(fmt.Sprintf("cruise.control.metrics.reporter.%s", k), v) -->  config.Set(fmt.Sprintf(cruiseControlConfigString, k), v).